### ISSUE TITLE
requires should generate "required='true'" attribute in Jackson @JsonProperty annotation

### ISSUE THREAD

**Issue by jmazzitelli** [joelittlejohn/jsonschema2pojo#429]
When using "requires" in a JSON schema, what is generated is a "(Required)" javadoc comment - for example:

   /**
   * (Required)
   */
   @JsonProperty("foo")

In the case of generating Jackson code, Jackson actually supports a "required" attribute in @JsonProperty. So this would be valid:

   @JsonProperty(value="foo", required=true)

How hard would it be to generate that code instead (if only for when generating Jackson - I don't know what other implementation support).

It is true that some versions of Jackson don't do anything with this annotation attribute, but having it would at least allow me to write my own reflection code and do the checks myself (I would be able to use reflection and find all getter methods annotations with that JsonProperty that has required=true and make sure it returns a non-null).

Without this annotation, and only with javadoc, there is no way for me to know what the schema flagged as required so I can't make this check.

I'd be willing to do the development of this, if you point me to the place in the code where this would need to be fixed (unless its easy enough for you to do it :))

**Comment by joelittlejohn** [comment]
Sounds good to me. I'll give it some thought but I don't see this as a particularly hard one.

**Comment by jmazzitelli** [comment]
I just submitted PR #430 - it works but only if you have "required" on the property in the schema. I wasn't sure how to get this to work if you only use the "required" array in the schema. See the comment in that PR for what I'm talking about. So this gets us half way there at least. But it would be nice to get that annotation attribute added if you use the "required" array in the schema (rather than having "required=true" in each property defined in the JSON schema - its nice to have a single "required" array in the JSON schema listing all the required attributes so with a single glance in the schema you can see what is required).

**Comment by jmazzitelli** [comment]
Can I ask the status of this issue and the related PR?

**Comment by joelittlejohn** [comment]
Hi @jmazzitelli. The PR isn't mergeable until we have support for both of the 'required' property formats I'm afraid. Having this work in only one case will be surprising and confusing for people, since both 'required' property formats are supported in other places. We just can't have half-complete features included as it makes it very difficult for user's to understand when something will or will not work.

I had a quick look at the changes you've made so far. One other thing that's essential before we can merge a PR is integration tests. Without these, the feature can be lost or broken at any point and we won't know about it.

**Comment by samskiter** [comment]
Hi, I'm here looking for required stuff too. the boolean required flag is from json schema draft v3 and the required array is from draft v3. I would consider the latter much  more likely to crop up, therefore.

Would you consider adding the "Nullable" field to all properties no in the "required" array if and only if the "required" array is present? This would generate a nice warning if you try to access a non-required field without checking for null.

**Comment by jmazzitelli** [comment]
FWIW: I tried to figure out how to implement this with the required array specified, but could not figure out how to do it. Which is why I left this half-implemented. Was hoping someone could take this work in the PR and run with it to complete it.

**Comment by blackmamo** [comment]
It is also worth noting that the @JsonProperty annotation on the field will not cause jackson to do any validation when deserialising. If you want that to be done, then you need the constructor to be annotated @JsonCreator, and then annotate the constructor arguments. Has that aspect of this request been considered?

**Comment by payne911** [comment]
Any status update on this?
